src/modules/url-changer.js

```diff
@@ -315,75 +170,106 @@
 		return result;
 	};
 	
-	var onUpdateQsFragment = function (key, data, e) {
-		if ($.isPlainObject(data)) {
-			//Update _currentQsFragment
-			$.extend(_currentQsFragment, data);
+	var onUpdateQsFragment = function (key, options, e) {
+		if ($.isPlainObject(options.qs)) {
+			//Update currentQsFragment
+			$.extend(currentQsFragment, options.qs);
 			
-			var currentQsIndex = _currentPageFragment.indexOf('?'),
-			newQsString = _generateQsString();
+			var currentQsIndex = currentPageFragment.indexOf('?'),
+			newQsString = generateQsString();
 			
-			if (newQsString !== _currentPageFragment) {
+			if (newQsString !== currentPageFragment) {
 				//Generate new page fragment
 				if (!newQsString.length) {
-					_currentPageFragment = '';
+
+					//Trim all existing qs
+					if (currentQsIndex !== -1) {
+						currentPageFragment = currentPageFragment.split('?')[0];
+					}
 				} else if (currentQsIndex === -1) {
-					_currentPageFragment += '?' + newQsString;
+					currentPageFragment += '?' + newQsString;
 				} else {
-					_currentPageFragment = _currentPageFragment.substring(0, currentQsIndex + 1) + 
+					currentPageFragment = currentPageFragment.substring(0, currentQsIndex + 1) +
 						newQsString;
 				}
 				
 				//_currentPage
-				_currentStrategy.updateUrlFragment();
-				$.sendPageView({page: data.route});
+				updateUrlFragment();
+
+				if (options.raiseFragmentChanged) {
+					App.mediator.notify('page.fragmentChanged', currentPageFragment);
+				}
+
+				//??? options.route doesnt exists....
```

Comment:
The parameter can be remove as the defaults will be fine since `updateUrlFragment()` does the job.